src/main/java/net/fabricmc/fabric/mixin/client/itemgroup/MixinItemGroup.java

```diff
@@ -0,0 +1,44 @@
+package net.fabricmc.fabric.mixin.client.itemgroup;
+
+import net.fabricmc.fabric.client.itemgroup.ItemGroupExtensions;
+import net.minecraft.item.ItemGroup;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ItemGroup.class)
+public abstract class MixinItemGroup implements ItemGroupExtensions {
+
+	@Shadow @Final @Mutable
+	public static ItemGroup[] GROUPS;
+
+	@Shadow public abstract int getId();
+
+	@Override
+	public void fabric_expandArray() {
+		ItemGroup[] tempGroups = GROUPS;
+		GROUPS = new ItemGroup[GROUPS.length + 1];
+		for(ItemGroup group : tempGroups){
```

Comment:
We could probably use System.arraycopy here, but at this size it doesn't matter.